lib/Hooks.php

```diff
@@ -260,8 +260,13 @@ public function handleFirstLogin(GenericEvent $event): void {
 			$this->shareManager->updateShare($share);
 		}
 
-		// Disable previous account
-		$guestUser->setEnabled(false);
+		if ($this->config->getSystemValue('remove_guest_account_on_conversion', false) === false) {
```

Comment:
I'd prefer an app config, but seeing that other settings are already in the config file, fine by me.